### PR TITLE
Add EACRuntime CRD

### LINKED ISSUE #2340: [FEATURES]Add EACRuntime CRD and generated code.
**What feature you'd like to add:**

**Why is this feature needed:**

### PR DESCRIPTION
**Pull request by myccccccc**
### Ⅰ. Describe what this PR does
Add EACRuntime CRD and generated code.

### Ⅱ. Does this pull request fix one issue?

fixes #2340 

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: config/crd/patches/webhook_in_eacruntimes.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  conversion:
+    strategy: Webhook
+    webhookClientConfig:
+      # this is "\n" used as a placeholder, otherwise it will be rejected by the apiserver for being blank,
+      # but we're going to set it later using the cert-manager (or potentially a patch if not using cert-manager)
+      caBundle: Cg==
+      service:
+        namespace: system
+        name: webhook-service
+        path: /convert
```
Comment (cheyang) [inline config/crd/patches/webhook_in_eacruntimes.yaml:17]: Add return of line.

File: api/v1alpha1/eacruntime_types.go [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	EACRuntimeKind = "EACRuntime"
+)
+
+// InitAlifuseSpec is a description of initialize the fuse kernel module for runtime
+type InitAlifuseSpec struct {
```
Comment (cheyang) [inline api/v1alpha1/eacruntime_types.go:29]: I suggest using `InitFuseSpec` instead.

File: api/v1alpha1/eacruntime_types.go [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	EACRuntimeKind = "EACRuntime"
```
Comment (cheyang) [inline api/v1alpha1/eacruntime_types.go:25]: Could you add comments about the meaning of EAC?

File: api/v1alpha1/eacruntime_types.go [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// Enabled or Disabled for the components.
+	// Default enable.
+	// +optional
+	Disabled bool `json:"disabled,omitempty"`
+
+	// NodeSelector is a selector which must be true for the master to fit on a node.
+	// +optional
```
Comment (cheyang) [inline api/v1alpha1/eacruntime_types.go:67]: change `master` to `component`.

File: api/v1alpha1/eacruntime_types.go [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+	// <br>
+	// Resources are not allowed for ephemeral containers. Ephemeral containers use spare resources
+	// already allocated to the pod.
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// If the fuse client should be deployed in global mode,
+	// otherwise the affinity should be considered
+	// +optional
+	Global bool `json:"global,omitempty"`
```
Comment (cheyang) [inline api/v1alpha1/eacruntime_types.go:95]: Global is deprecated. Please remove it. thanks.

File: api/v1alpha1/eacruntime_types.go [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+	// Tiered storage used by EAC worker
+	TieredStore TieredStore `json:"tieredstore,omitempty"`
+
+	// The replicas of the worker, need to be specified
+	Replicas int32 `json:"replicas,omitempty"`
+
+	// Enable OS other than AliyunOS
+	// Non-AliyunOS is not enabled by default
+	// +optional
+	EnableNoneAliyunOS bool `json:"enableNoneAliyunOS,omitempty"`
```
Comment (cheyang) [inline api/v1alpha1/eacruntime_types.go:139]: I suggest using `OperationSystemOptimization` struct.

```yaml
type OperationSystemOptimization struct {
      OSVersion string `json:"osVersion,omitempty"`
       Enabled bool `json:"enabled,omitempty"`
}
```

File: api/v1alpha1/eacruntime_types.go [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+
+	// Aliyun AccessKey ID for DirQuota API
```
Comment (cheyang) [inline api/v1alpha1/eacruntime_types.go:141]: I don't suggest using AccessKeyID here. Instead, you can using the encrypted  option in the dataset.

**Comment by myccccccc** [comment]
@cheyang Your comments have been addressed, PTAL again.

File: api/v1alpha1/eacruntime_types.go [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+	Fuse EACFuseSpec `json:"fuse,omitempty"`
+
+	// Tiered storage used by EAC worker
+	TieredStore TieredStore `json:"tieredstore,omitempty"`
+
+	// The replicas of the worker, need to be specified
+	Replicas int32 `json:"replicas,omitempty"`
+
+	// Operation system optimization for EAC
+	OperationSystemOptimization OperationSystemOptimization `json:"operationSystemOptimization,omitempty"`
```
Comment (cheyang) [inline api/v1alpha1/eacruntime_types.go:145]: you can name it osAdvise for easy to use.

File: api/v1alpha1/eacruntime_types.go [hunk 9]
```diff
@@ -142,7 +142,7 @@ type EACRuntimeSpec struct {
 	Replicas int32 `json:"replicas,omitempty"`
 
 	// Operation system optimization for EAC
-	OperationSystemOptimization OperationSystemOptimization `json:"operationSystemOptimization,omitempty"`
+	OSAdvise OperationSystemOptimization `json:"osAdvise,omitempty"`
```
Comment (cheyang) [inline api/v1alpha1/eacruntime_types.go:145]: Can you rename OperationSystemOptimization to OSAdvise?

Comment (myccccccc) [reply api/v1alpha1/eacruntime_types.go]: done

Comment (myccccccc) [reply api/v1alpha1/eacruntime_types.go]: > Can you rename OperationSystemOptimization to OSAdvise?

done

**Review by cheyang — APPROVED** [review]

**Comment by cheyang** [comment]
Thanks, @myccccccc !

**Pull request merged**